llvm/lib/CodeGen/AsmPrinter/DwarfUnit.cpp

```diff
@@ -1777,6 +1779,10 @@ void DwarfUnit::emitCommonHeader(bool UseOffsets, dwarf::UnitType UT) {
 }
 
 void DwarfTypeUnit::emitHeader(bool UseOffsets) {
+  if (!DD->useSplitDwarf()) {
+    LabelBegin = Asm->createTempSymbol("tu_begin");
+    Asm->OutStreamer->emitLabel(LabelBegin);
+  }
```

Comment:
it's probably not especially expensive to just always label the DwarfTypeUnits

Reply:
I feel like I got an assert without the check. Also there are no relocations done on the .dwo sections, so kind of weird to have a symbol there no?

Reply:
Fair enough - there's not harm in putting local symbols in dwo sections, since they won't be emitted into dwo files anyway - but yeah, no reason to.

Knowing what the assert was might be interesting. (perhaps we do make the simplifying assumption and assert that there aren't labels in dwo sections, but I don't think so... - I think we do assert that there aren't any /relocations/ in dwo files)

Reply:
OK I was wrong. I can't repro assert anymore. Not sure how I was able to trigger it previously.